app/layersproxymodel.cpp

```diff
@@ -13,22 +13,10 @@
 #include "qgsproject.h"
 #include "qgslayertree.h"
 
-LayersProxyModel::LayersProxyModel( LayersModel *model, LayerModelTypes modelType ) :
-  mModelType( modelType ),
-  mModel( model )
+LayersProxyModel::LayersProxyModel( QObject *parent ) :
+  QgsMapLayerProxyModel{ parent },
+  filterFunction( []( QgsMapLayer * layer ) { return ( layer && layer->isValid() ); } ) // default filter
```

Comment:
Would it be possible to assign this default behaviour directly in the `layersproxymodel.h`?

Reply:
It's quite complicated to read here

Reply:
Or optionally just call the `applyFilterFunction` method directly in the constructor as previously. It will work, because now you also "reset" the filtering function when the model type changes